### PR TITLE
feat(dropdown): add error and empty states templates

### PR DESCRIPTION
**Pull request by saty2103**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/atoms/dropdown/__stories__/variants/WithErrorTempate.story.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          searchTerm,
+          options: searchedOptions,
+          count: searchedOptions.length,
+        });
+      }, 1000);
+    });
+  };
+
+  const errorTemplate = <Heading appearance={'disabled'}>No data is fetched.</Heading>;
+
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/__stories__/variants/WithErrorTempate.story.jsx]: Please use the proper error template as provided in design

File: core/components/atoms/dropdown/DropdownList.tsx [hunk 2]
```diff
@@ -244,6 +253,8 @@ const DropdownList = (props: OptionsProps) => {
     minHeight && setMinHeight(minHeight);
   };
 
+  const _isNoResultReturned = listOptions.length === 0 && !loadingOptions && selected.length <= 0;
+
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/DropdownList.tsx]: Please change this variable name

File: core/components/atoms/dropdown/__stories__/Options.tsx [hunk 3]
```diff
@@ -224,3 +224,16 @@ export const fetchOptions = (searchTerm: string, options = dropdownOptions) => {
...
   });
 };
+
+export const fetchOptionsEmpty = (searchTerm: string, options = []) => {
+  const searchedOptions = searchTerm ? getSearchedOptions(options, searchTerm) : options;
+  return new Promise<fetchOptionSchema>((resolve) => {
+    window.setTimeout(() => {
+      resolve({
+        searchTerm,
+        options: searchedOptions,
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/__stories__/Options.tsx]: Why do we need to call `getSearchedOptions` function when we can directly pass empty array to `options: []`

File: core/components/atoms/dropdown/__stories__/Options.tsx [hunk 4]
```diff
@@ -224,3 +224,16 @@ export const fetchOptions = (searchTerm: string, options = dropdownOptions) => {
     }, 1000);
   });
 };
+
+export const fetchOptionsEmpty = (searchTerm: string, options = []) => {
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/__stories__/Options.tsx]: Please change the function name. It should be like `fetchEmptyOptions` or something similar to this

File: core/components/atoms/dropdown/__stories__/variants/WithErrorTempate.story.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import { Uncontrolled, Controlled } from '../_common_/types';
+
+// CSF format story
+export const withErrorTemplate = () => {
+  const options = [];
+
+  const getSearchedOptions = (options, searchTerm) => {
+    const result = options.filter((option) => option.label.toLowerCase().includes(searchTerm.toLowerCase()));
+    return result;
+  };
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/__stories__/variants/WithErrorTempate.story.jsx]: When `options` declared above is always empty then this function always return the empty array. There is no need to call this function when we can directly pass empty array to promise

File: core/components/atoms/dropdown/__tests__/Dropdown.test.tsx [hunk 6]
```diff
@@ -486,6 +487,29 @@ describe('Dropdown component with search', () => {
     fireEvent.click(dropdownTrigger);
     expect(screen.getByPlaceholderText('Custom search text')).toBeInTheDocument();
   });
+
+  it('when no record is available', () => {
+    const { getByTestId, queryByTestId } = render(<Dropdown options={undefined} withSearch={true} />);
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/__tests__/Dropdown.test.tsx]: Please write proper description for the test case. Also please do not pass undefined to the options

File: core/components/atoms/dropdown/errorTemplate.tsx
Comment (anuradha9712) [inline core/components/atoms/dropdown/errorTemplate.tsx:1]: This file name should be `ErrorTemplate.tsx`

File: core/components/atoms/dropdown/errorTemplate.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { Text } from '@/index';
+
+interface ErrorTemplateProps {
+  dropdownStyle: React.CSSProperties;
+  width: number | string | undefined;
+  errorType: string;
+}
+
+const errorTemplateHeader: Record<string, string> = {
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/errorTemplate.tsx]: This variable name should be `errorTemplateTitle`

File: core/components/atoms/dropdown/DropdownList.tsx [hunk 8]
```diff
@@ -679,6 +700,8 @@ const DropdownList = (props: OptionsProps) => {
     }
   };
 
+  const shouldSearchRender = (withSearch || props.async) && (!_isNoResultReturned || errorType === 'NO_RECORDS_FOUND');
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/DropdownList.tsx]: Please change this variable name

**Comment by anuradha9712** [comment]
Please change the commit message to "feat(dropdown): add error and empty states templates"

File: core/components/atoms/dropdown/errorTemplate.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  NO_RECORDS_FOUND: 'Try modifying your search to find what you are looking for.',
+  DEFAULT: 'We have nothing to show you at the moment.',
+};
+
+export const ErrorTemplate: React.FC<ErrorTemplateProps> = ({ dropdownStyle, width, errorType }) => {
+  return (
+    <div className="Dropdown-wrapper px-7 d-flex" style={dropdownStyle} data-test="DesignSystem-Dropdown--wrapper">
+      <div
+        className={'Dropdown--errorWrapper'}
+        style={{ width: `calc(${width}px - 2 * var(--spacing-xl))` }}
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/errorTemplate.tsx]: There is no need to pass this style tag

File: core/components/atoms/dropdown/errorTemplate.tsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        data-test="DesignSystem-Dropdown--errorWrapper"
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/errorTemplate.tsx]: Same data-test id already exists. Please use something other name

File: css/src/components/dropdown.css [hunk 11]
```diff
@@ -56,6 +56,13 @@
   box-sizing: border-box;
 }
 
+.Dropdown--errorWrapper {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+}
```
Comment (anuradha9712) [inline css/src/components/dropdown.css]: All these classes already exists in the design system. Please remove this

File: core/components/atoms/dropdown/errorTemplate.tsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { Text } from '@/index';
+
+interface ErrorTemplateProps {
+  dropdownStyle: React.CSSProperties;
+  width: number | string | undefined;
+  errorType: string;
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/errorTemplate.tsx]: type of `errorType` should not be `string` here, as it only accepts `FAILED_TO_FETCH` | `NO_RECORDS_FOUND` | `DEFAULT`

File: core/components/atoms/dropdown/DropdownList.tsx (hunk 2 above)
Comment (saty2103) [reply core/components/atoms/dropdown/DropdownList.tsx]: Variable name changed to **isDropdownListBlank**.

File: core/components/atoms/dropdown/DropdownList.tsx (hunk 8 above)
Comment (saty2103) [reply core/components/atoms/dropdown/DropdownList.tsx]: variable name changed to **enableSearch**.

File: core/components/atoms/dropdown/__stories__/Options.tsx (hunk 4 above)
Comment (saty2103) [reply core/components/atoms/dropdown/__stories__/Options.tsx]: Done

File: core/components/atoms/dropdown/errorTemplate.tsx
Comment (saty2103) [reply core/components/atoms/dropdown/errorTemplate.tsx:1]: Done

File: core/components/atoms/dropdown/errorTemplate.tsx (hunk 12 above)
Comment (saty2103) [reply core/components/atoms/dropdown/errorTemplate.tsx]: Done

File: core/components/atoms/dropdown/errorTemplate.tsx (hunk 7 above)
Comment (saty2103) [reply core/components/atoms/dropdown/errorTemplate.tsx]: Done

File: css/src/components/dropdown.css (hunk 11 above)
Comment (saty2103) [reply css/src/components/dropdown.css]: Done

File: core/components/atoms/dropdown/errorTemplate.tsx (hunk 9 above)
Comment (saty2103) [reply core/components/atoms/dropdown/errorTemplate.tsx]: Done

File: core/components/atoms/dropdown/__stories__/variants/WithErrorTempate.story.jsx (hunk 1 above)
Comment (saty2103) [reply core/components/atoms/dropdown/__stories__/variants/WithErrorTempate.story.jsx]: Done

**Pull request closed without merging**